seqr/utils/file_utils.py

```diff
@@ -64,3 +64,25 @@ def _google_bucket_file_iter(gs_path, byte_range=None, raw_content=False, user=N
             line = line.decode('utf-8')
         yield line
 
+
+def get_vcf_filename(vcf_path):
+    if vcf_path.endswith('.vcf') or vcf_path.endswith('.vcf.gz') or vcf_path.endswith('.vcf.bgz'):
+        return vcf_path
+    process = subprocess.Popen('gsutil ls ' + vcf_path, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
+    if process.wait() != 0:
+        return None
+    for line in process.stdout:
+        file = line.decode('utf-8').strip()
+        if file.endswith('.vcf') or file.endswith('.vcf.gz') or file.endswith('.vcf.bgz'):
+            return file
+    return None
```

Comment:
Must we do the mapping from the data path (`Path to the Joint Called VCF`) to VCF file name mapping? Can we ask the user to provide the VCF filename directly?

Reply:
I'm really confused what filename we are even getting here? In the main code block, we have 
```
data_path = 'gs://{bucket}/{path}'.format(bucket=bucket_name.rstrip('/'), path=request_json['dataPath'].lstrip('/'))
if not does_file_exist(data_path, user=request.user):
    error = 'Data file or path {} is not found.'.format(request_json['dataPath'])
    return create_json_response({'error': error}, status=400, reason=error)
```
where `does_file_exist` is ultimately running a `gsutil ls` check on that full `data_path` and throws an error if it fails. So I don't get what we are checking here? In what scenario is the `file` that is returned from this function different from the `data_path` input into it?

Reply:
To my understanding, the path that the user provides is just the path to the VCF file, not including the filename. Especially, the `data_path` is different from the `filename` when the VCF is split into multiple files. In that case, the `data_path` is a directory, not a file. I know the VCF can be split per chromosome basis and we can use the samples in any one of the non-Y chromosome splits of the VCF. I'm not sure if there are any other ways for splitting VCF.

I could combine the path existing check and getting a VCF filename so that there would be only one `gsutil ls` would be run.

Reply:
We do not support sharded VCFs for anvil loading so the split file thing is not an edge case we need to hand;e

Reply:
If we don't support split VCF, should I also update the helper message on the data loading page as below?
![image](https://user-images.githubusercontent.com/33550059/128737573-b2b7f56c-377d-4222-8639-7286eb81489d.png)